types/logfmt/index.d.ts

```diff
@@ -0,0 +1,66 @@
+// Type definitions for logfmt 1.2
+// Project: https://github.com/csquared/node-logfmt#readme
+// Definitions by: Evan Broder <https://github.com/ebroder>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+/// <reference types="node" />
+
+import { IncomingMessage, ServerResponse } from 'http';
+
+type HTTPHandler =
+  (req: IncomingMessage, res: ServerResponse, next: (err?: any) => void) => any;
+
+interface WritableStream {
+    write(data: string): void;
+}
+
+interface RequestLoggerOptions {
+    immediate?: boolean;
+    elapsed?: string;
+}
+type RequestLoggerFormatter =
+  (req: IncomingMessage, res: ServerResponse) => object;
+
+interface RequestLogger {
+    (options?: RequestLoggerOptions, formatter?: RequestLoggerFormatter): HTTPHandler;
+    (formatter: RequestLoggerFormatter): HTTPHandler;
+    commonFormatter: (req: IncomingMessage, res: ServerResponse) => {
+        ip: string;
+        time: string;
+        method: string;
+        path: string;
+        "status": number;
+        request_id?: string;
+        content_length?: string;
+        content_type?: string;
+    };
+}
+
+declare class Logfmt {
+    stringify(data: object): string;
+    parse(line: string): object;
+
+    log(data?: object, stream?: WritableStream): void;
+    time(label?: string): Logfmt;
+    namespace(data: object): Logfmt;
+    error(err: Error, id?: string): void;
+
+    streamParser(options?: { end?: boolean; }): NodeJS.ReadWriteStream;
+    streamStringify(options?: { delimiter?: string }): NodeJS.ReadWriteStream;
+
+    bodyParser(options?: { contentType?: string }): HTTPHandler;
+    bodyParserStream(options?: { contentType?: string }): HTTPHandler;
+
+    requestLogger: RequestLogger;
+
+    stream: WritableStream;
+    maxErrorLines: number;
+}
+
+// logfmt copies its prototype onto the exported class, so you can both use it
+// directly as an instance of the class and as a constructor to instantiate new
+// instances.
+declare const logfmt: Logfmt & typeof Logfmt;
```

Comment:
Sorry about the delay, but thanks for your patience.

Instead of doing this, how about

```ts
interface Logfmt {
    stringify(data: object): string;
    parse(line: string): object;

    log(data?: object, stream?: WritableStream): void;
    time(label?: string): Logfmt;
    namespace(data: object): Logfmt;
    error(err: Error, id?: string): void;

    streamParser(options?: { end?: boolean; }): NodeJS.ReadWriteStream;
    streamStringify(options?: { delimiter?: string }): NodeJS.ReadWriteStream;

    bodyParser(options?: { contentType?: string }): HTTPHandler;
    bodyParserStream(options?: { contentType?: string }): HTTPHandler;

    requestLogger: RequestLogger;

    stream: WritableStream;
    maxErrorLines: number;
}

interface LogfmtStatic extends Logfmt {
    new (): Logfmt;
}

declare const logfmt: LogfmtStatic;
```

Reply:
Sure, that seems much cleaner :slightly_smiling_face:

Pushed a fix